Objects/interpolationobject.c

```diff
@@ -0,0 +1,217 @@
+}
+
+static void
+interpolation_dealloc(interpolationobject *self)
+{
+    PyObject_GC_UnTrack(self);
+    Py_CLEAR(self->value);
+    Py_CLEAR(self->expression);
+    Py_CLEAR(self->conversion);
+    Py_CLEAR(self->format_spec);
+    Py_TYPE(self)->tp_free(self);
+}
+
+static int
+interpolation_traverse(interpolationobject *self, visitproc visit, void *arg)
+{
+    Py_VISIT(self->value);
+    Py_VISIT(self->expression);
+    Py_VISIT(self->conversion);
+    Py_VISIT(self->format_spec);
+    return 0;
+}
+
+static PyObject *
+interpolation_repr(interpolationobject *self)
+{
+    return PyUnicode_FromFormat("%s(%R, %R, %R, %R)",
+                                _PyType_Name(Py_TYPE(self)),
+                                self->value, self->expression,
+                                self->conversion, self->format_spec);
+}
+
+static PyMemberDef interpolation_members[] = {
+    {"value", Py_T_OBJECT_EX, offsetof(interpolationobject, value), Py_READONLY, "Value"},
+    {"expression", Py_T_OBJECT_EX, offsetof(interpolationobject, expression), Py_READONLY, "Expression"},
+    {"conversion", Py_T_OBJECT_EX, offsetof(interpolationobject, conversion), Py_READONLY, "Conversion"},
+    {"format_spec", Py_T_OBJECT_EX, offsetof(interpolationobject, format_spec), Py_READONLY, "Format specifier"},
+    {NULL}
+};
+
+PyTypeObject _PyInterpolation_Type = {
+    PyVarObject_HEAD_INIT(NULL, 0)
+    .tp_name = "string.templatelib.Interpolation",
+    .tp_doc = PyDoc_STR("Interpolation object"),
+    .tp_basicsize = sizeof(interpolationobject),
+    .tp_itemsize = 0,
+    .tp_flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_HAVE_GC,
+    .tp_new = (newfunc) interpolation_new,
+    .tp_alloc = PyType_GenericAlloc,
+    .tp_dealloc = (destructor) interpolation_dealloc,
+    .tp_free = PyObject_GC_Del,
+    .tp_repr = (reprfunc) interpolation_repr,
+    .tp_members = interpolation_members,
+    .tp_traverse = (traverseproc) interpolation_traverse,
+};
+
+static PyObject *
+_get_match_args(void)
+{
+    PyObject *value = NULL, *expression = NULL, *conversion = NULL, *format_spec = NULL;
+    PyObject *tuple = NULL;
+
+    value = PyUnicode_FromString("value");
+    if (!value) {
+        goto error;
+    }
+    expression = PyUnicode_FromString("expression");
+    if (!expression) {
+        goto error;
+    }
+    conversion = PyUnicode_FromString("conversion");
+    if (!conversion) {
+        goto error;
+    }
+    format_spec = PyUnicode_FromString("format_spec");
+    if (!format_spec) {
+        goto error;
+    }
+
+    tuple = PyTuple_Pack(4, value, expression, conversion, format_spec);
+
+error:
+    Py_XDECREF(value);
+    Py_XDECREF(expression);
+    Py_XDECREF(conversion);
+    Py_XDECREF(format_spec);
+    return tuple;
+
+}
+
+PyStatus
+_PyInterpolation_InitTypes(PyInterpreterState *interp)
+{
+    PyObject *tuple = _get_match_args();
+    if (!tuple) {
+        goto error;
+    }
+
+    int status = PyDict_SetItemString(_PyType_GetDict(&_PyInterpolation_Type),
```

Comment:
`_PyType_GetDict` can also return `NULL`, please add a check for it:

```suggestion
    PyObject *d = _PyType_GetDict(&_PyInterpolation_Type);
    if (d == NULL) {
        goto error;
    }
    int status = PyDict_SetItemString(d,
```